pyfar/dsp/dsp_interpolation.py

```diff
@@ -0,0 +1,731 @@
+
+    Parameters
+    ----------
+    signal : pyfar.Signal
+        The input data.
+    width : number
+        The width of the smoothing window in octaves, e.g., 1/3 will apply
+        third octave smoothing.
+    mode : str, optional
+
+        ``"magnitude"``
+            Only the magnitude response, i.e., the absolute spectrum is
+            smoothed. Note that this return as zero-phase signal. It might be
+            necessary to generate a minimum or linear phase if the data is
+            subject to further processing after the smoothing (cf.
+            :py:func:`~pyfar.dsp.minimum_phase` and
+            :py:func:`~pyfar.dsp.linear_phase`)
+        ``"magnitude_phase"``
+            Separately smooth the magnitude and unwrapped phase response
+        ``"magnitude_copy"``
+            Smooth the magnitude and keep the phase of the input signal.
+        ``"complex"``
+            Separately smooth the real and imaginary part of the spectrum. This
+            method often causes artifacts at high frequencies.
+
+        The default is ``"magnitude"``.
+    window : str, optional
+        String that defines the smoothing window. All window from
+        :py:func:`~pyfar.dsp.time_window` that does not require an additional
+        parameter can be used. The default is "boxcar", which uses the
+        most commonly used rectangular window.
+
+    Returns
+    -------
+    signal : pyfar.Signal
+        The smoothed output data
+    window_stats : tuple
+        A tuple containing information about the smoothing process
+
+        `n_window`
+            The window length in (logarithmically spaced) samples
+        `window_width`
+            The actual window width in octaves. This can deviate due to the
+            ideal width because the window must have an integer length
+
+
+    References
+    ----------
+    .. [#] J. G. Tylka, B. B. Boren, and E. Y. Choueiri, “A Generalized Method
+           for Fractional-Octave Smoothing of Transfer Functions that Preserves
+           Log-Frequency Symmetry (Engineering Report),” J. Audio Eng. Soc. 65,
+           239-245 (2017). doi:10.17743/jaes.2016.0053
+
+    Examples
+    --------
+
+    Octave smoothing of continuous spectrum consiting of two bell filters.
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>>
+        >>> signal = pf.signals.impulse(441)
+        >>> signal = pf.dsp.filter.bell(signal, 1e3, 12, 1, "III")
+        >>> signal = pf.dsp.filter.bell(signal, 10e3, -60, 100, "III")
+        >>>
+        >>> smoothed, _ = pf.dsp.smooth_fractional_octave(signal, 1)
+        >>>
+        >>> ax = pf.plot.freq(signal, label="input")
+        >>> pf.plot.freq(smoothed, label="smoothed")
+        >>> ax.legend(loc=3)
+
+    Octave smoothing of the discontinuous spectrum of a sine signal causes
+    artifacts at the edges if using method 2 (cf. Tylka et al. 2017 Fig. 4).
+    This is caused be the intermediate interpolation steps described above.
```

Comment:
I think mentioning "method 2" without knowing the details from Tylka is a little confusing here. Instead maybe say something like "this method" and add a short comment, that the method is also not generally meant to smooth such signals as sine?

Reply:
How about this?

Reply:
yes, that's good :)